apps/andi/test/integration/andi_web/controllers/dataset_controller_test.exs

```diff
@@ -356,6 +356,14 @@ defmodule Andi.CreateDatasetTest do
     end
   end
 
+  describe "dataset get" do
+    test "andi doesn't return server in response headers" do
+      {:ok, %Tesla.Env{headers: headers}} = get("/api/v1/datasets", headers: [{"content-type", "application/json"}])
```

Comment:
The test itself is fine, but I would suggest either pattern matching out the tuple you care about (assuming you know exactly what headers are coming back) or turning it into a map and checking if it has the `server` key:
```suggestion
      {:ok, %Tesla.Env{headers: headers}} = get("/api/v1/datasets", headers: [{"content-type", "application/json"}])
      refute headers |> Map.new() |> Map.has_key?("server")
```